pydantic/dataclasses.py

```diff
@@ -106,6 +106,27 @@ def _pydantic_post_init(self: 'Dataclass', *initvars: Any) -> None:
     return _pydantic_post_init
 
 
+def _generate_builtin_dataclass_wrapper_name(cls: Type[Any]) -> str:
+    """
+    Generate a name for the class that wraps a built-in dataclass that is:
+    - a valid Python identifier (usable as attribute name),
+    - unique (no collisions with different classes),
+    - deterministic (so it works across different processes), and
+    - recognisable as wrapper for the dataclass (to ease debugging).
+    """
+
+    # Assumption: the combination of `__module__` and `__qualname__` attributes
+    # uniquely identifies the class. This should be a safe assumption because pickle
+    # uses these two attributes to find the class.
+    fully_qualified_cls_name = f'{cls.__module__}.{cls.__qualname__}'
+
+    # We hex-encode the fully qualified name of the source class because it contains
+    # one or more period (.) characters, making it unusable as identifier.
+    unique_id = fully_qualified_cls_name.encode('utf-8').hex()
```

Comment:
do we really need to use hex and make the name (arguably) harder to read?

Could we use just `f'{cls.__module__}ᚑ{cls.__qualname__}'`? I know that means in theory that we could get clashes, but it seems vanishingly unlike unless someone is trying to clash.

(the `ᚑ` is a arbitarily selected high order unicode character which is [valid in identifiers](https://gist.github.com/samuelcolvin/68f56c96bf18b03bb8a3ffa988380966), we could use something else if you prefer)